server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -498,7 +497,7 @@ UploadedMetadataResults writeMetadataInParallel(
             + (uploadDiscoveryNodes ? 1 : 0) + (uploadClusterBlock ? 1 : 0) + (uploadTransientSettingMetadata ? 1 : 0)
             + clusterStateCustomToUpload.size() + (uploadHashesOfConsistentSettings ? 1 : 0) + indicesRoutingToUpload.size();
         CountDownLatch latch = new CountDownLatch(totalUploadTasks);
-        Map<String, CheckedRunnable<IOException>> uploadTasks = new ConcurrentHashMap<>(totalUploadTasks);
+        List<String> uploadTasks = Collections.synchronizedList(new ArrayList<>(totalUploadTasks));
```

Comment:
Can we add some validation at the end to verify that the size of results is the same as that of uploadTasks ?

Reply:
Added a validation after upload